Source/DafnyCore/AST/Expressions.cs

```diff
@@ -808,6 +808,8 @@ public abstract class Expression {
     var le = this as StringLiteralExpr;
     return le == null ? null : le.Value as string;
   }
+
+  public virtual IEnumerable<INode> Children => SubExpressions;
```

Comment:
In my formatting PR, I am adding a field PreResolveSubExpressions for every ConcreteExpression (the expression that support rewriting), because I'm doing the formatting based on that tree and not the resolved one.
I had to duplicate several SubExpressions because they were slightly different.

Are your Children pre-resolve sub expressions? If so I'd suggest you rename `Children` explicitly `PreResolveChildren` or something like that, so that it's unambiguous which nodes you will be referring to.
And that will help me in my PR too.

Reply:
They're not pre-resolve, they're resolved.